### ISSUE TITLE
haproxy 2.2.2 cpu cores locked at 100%

### ISSUE THREAD

**Issue by ac-obair** [haproxy/haproxy#1005]
## Detailed description of the problem
The upgrade from 1.8.26 to 2.2.2 on centos 7.9.2009 seeing cpu core utilisation intermittently locked at 100%. This happens for 2 or 3 cores at a time and stays locked until haproxy is restarted. It takes a day or two for the cores to lock and it's never the same cores, seems random. 

This is an 8 core vm running on vmware. The utilisation issue reported looks to be the nbproc pid, 14406. We get roughly 5 to 6 milliion hits a day and this wasn't causing any notable drops in customer experience however finally all the healthchecks reported OK and their return codes all started returning as 502, obviously dropping service. Here core 8 and core 2 at 100%. I believe the nbproc pid was on core 2. 
<img width="1678" alt="Screenshot 2020-12-15 at 11 40 19" src="https://user-images.githubusercontent.com/20089429/102222750-be83ca00-3edb-11eb-895d-9b391355e0b0.png">
After restart
<img width="1622" alt="Screenshot 2020-12-15 at 13 29 12" src="https://user-images.githubusercontent.com/20089429/102222757-c2175100-3edb-11eb-83f8-15af3cb9defb.png">

I managed to get some strace output before the service dropped, however it's hard to know what to look for. Essentially I should see most of the load on nbproc pid but I can't explain why it's 100 or why there's two of them. These are a pair of proxies, the above image is the live server, the standby proxy which has not had any load since it's last reboot, essentially unused apart from health checks has the same locked 100% issue. 
 
As mentioned with the strange healthcheck behaviour, I do wonder if it's related. The main change from 1.8 to 2.2 in our config has been the healthcheck syntax updates, I don't know enough to know if that could cause this issue. However they all appear to be configured correctly 

## Expected behavior
2 or 3 cores don't locked to 100%

## Steps to reproduce the behavior

Not sure
1. 
2. 
3. 

## Do you have any idea what may have caused this?
The upgrade from 1.8.26 to 2.2.2 was part of it, whether our cpu config or something on the server is misconfigured I don't know.

## Do you have an idea how to solve the issue?
Restarting haproxy, reload doesn't release the pids. 

## What is your configuration?

The relevant config would be:
```
  daemon
  nbproc 1
  nbthread 8
  cpu-map auto:1/1-8 0-7
  spread-checks 4
```
and health check example
```
  cookie SERVERID insert indirect nocache
  hash-type consistent
  retries 5
  option redispatch
  option httpchk
  http-check send meth GET uri /something ver HTTP/1.1 hdr host www.com hdr user-agent HAProxy
  http-check expect status 200
  default-server inter 5s fastinter 2s downinter 2s
  server web01.a.com web01.a.com:80 check maxconn 70
```
```
 haproxy -f  /etc/haproxy/haproxy.cfg -c
Configuration file is valid
```

I can give more config if it's needed.
## Output of `haproxy -vv` and `uname -a`

```
HA-Proxy version 2.2.2 2020/07/31 - https://haproxy.org/
Status: long-term supported branch - will stop receiving fixes around Q2 2025.
Known bugs: http://www.haproxy.org/bugs/bugs-2.2.2.html
Running on: Linux 3.10.0-1160.6.1.el7.x86_64 #1 SMP Tue Nov 17 13:59:11 UTC 2020 x86_64
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = gcc
  CFLAGS  = -O2 -g -Wall -Wextra -Wdeclaration-after-statement -fwrapv -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wno-missing-field-initializers -Wtype-limits
  OPTIONS = USE_PCRE2=1 USE_LINUX_TPROXY=1 USE_CRYPT_H=1 USE_GETADDRINFO=1 USE_OPENSSL=1 USE_ZLIB=1 USE_SYSTEMD=1

Feature list : +EPOLL -KQUEUE +NETFILTER -PCRE -PCRE_JIT +PCRE2 -PCRE2_JIT +POLL -PRIVATE_CACHE +THREAD -PTHREAD_PSHARED +BACKTRACE -STATIC_PCRE -STATIC_PCRE2 +TPROXY +LINUX_TPROXY +LINUX_SPLICE +LIBCRYPT +CRYPT_H +GETADDRINFO +OPENSSL -LUA +FUTEX +ACCEPT4 +ZLIB -SLZ +CPU_AFFINITY +TFO +NS +DL +RT -DEVICEATLAS -51DEGREES -WURFL +SYSTEMD -OBSOLETE_LINKER +PRCTL +THREAD_DUMP -EVPORTS

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=8).
Built with OpenSSL version : OpenSSL 1.0.2k-fips  26 Jan 2017
Running on OpenSSL version : OpenSSL 1.0.2k-fips  26 Jan 2017
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : SSLv3 TLSv1.0 TLSv1.1 TLSv1.2
Built with network namespace support.
Built with zlib version : 1.2.7
Running on zlib version : 1.2.7
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE2 version : 10.23 2017-02-14
PCRE2 library supports JIT : no (USE_PCRE2_JIT not set)
Encrypted password support via crypt(3): yes
Built with gcc compiler version 4.8.5 20150623 (Red Hat 4.8.5-39)
Built with the Prometheus exporter as a service

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
            fcgi : mode=HTTP       side=BE        mux=FCGI
       <default> : mode=HTTP       side=FE|BE     mux=H1
              h2 : mode=HTTP       side=FE|BE     mux=H2
       <default> : mode=TCP        side=FE|BE     mux=PASS

Available services :
	prometheus-exporter

Available filters :
	[SPOE] spoe
	[COMP] compression
	[TRACE] trace
	[CACHE] cache
	[FCGI] fcgi-app
```

## If HAProxy crashed: Last outputs and backtraces

```
(paste your output here)
```

## Additional information (if helpful)
The boxes are all automated, they've had every stable version of haproxy installed since 1.5, this is the first time I'm seeing this issue. I noticed it a few days after the upgrade (about a month ago) and trieid a few different configs, the issue persisted. The boxes at that time has 4 cores and have upgraded to 8 due to demand on the servers happening regardless of the haproxy issue obviously the issue persisted.

**Comment by capflam** [comment]
First of all, try to upgrade. A significant amount of bugs were fixed since the 2.2.2. Try the 2.2.6 please.

**Comment by ac-obair** [comment]
Ok happy to close this until I've done that testing.
